pkg/api/message/v1/dispatcher.go

```diff
@@ -79,6 +79,43 @@ func (d *dispatcher) Unregister(ch chan interface{}, topics ...string) {
 	d.unregister(ch, topics...)
 }
 
+// Update takes in an array of topics and unregisters any topics that are not in the array and registers any topics that are not already registered.
+func (d *dispatcher) Update(ch chan interface{}, topics ...string) {
```

Comment:
I don't strictly need to create this method. I _could_ just unregister everything and then re-register everything.

I added this method for two reasons:
1. Messages could get lost if a message got published during the unregister process and was waiting for the lock
2. For large numbers of subs (we have people subscribing to 10k+ topics at a time), it's going to be a lot more expensive to add and remove all those listeners.

I could be convinced that I'm over-thinking this and we should just simplify.

Reply:
I think we definitely do want the smart update behavior, the only thing I'm wondering if it's worth having add/remove subscriptions request, given that the lists could be 10k+ long.

Reply:
It's a reasonable question. It's nice in that there is less data going over the wire. But it does put more work onto the client to maintain a list of subscriptions and figure out the delta. Because there is no ackowledgement that an update has succeeded, this way also gives us a "self-healing" property where if some update isn't successfully processed the next update will set things right.